### ISSUE TITLE
Add temperature and salinity profile plot types

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#241]

**Comment by j-harbin** [comment]
Hi All,

I am looking for a vote on the proper syntax for plotting a profile. The options are as follows:

1. plot(x, which='profile', xtype='temperature')
2. plotProfile(x, type='temperature')
3. plot(x, which='temperature profile')

My vote is as follows:
1. 0.3
2. 0
3. 0.7

**Comment by dankelley** [comment]
My vote is 
1. 0.1 ... a low score because this has two arguments to specify one thing.
2. 0 ... a zero score because I want users to find all plotting actions within `plot()`.
3. 0.9 .. a high score because this is the phrase I'd use to describe such a plot.

**Comment by richardsc** [comment]
I know I've probably missed some discussion on this, but doesn't it make sense to tie this in with `plotProfile()` in `oce`? Otherwise we'll end up with two similar functions that do slightly different things.

I don't think I'm ready to vote until we've discussed the idea a bit more (perhaps tomorrow at the vf2f?) but I have to say I don't like all the typing in option 3 all that much ...

**Comment by j-harbin** [comment]
Thanks for the feedback everyone. As discussed in a vftf with @dankelley and @richardsc, the final decision was to do the following syntax:

plot(argos, which='profile', parameter='x')

This allows for freedom when choosing parameter type, which can be particularly interesting for BGC Argo. I am currently working on this in branch issue241.

**Comment by dankelley** [comment]
This also lets us add `plotProfile()` if we want to, as a one- or two-liner that calls `plot()`.

**Comment by j-harbin** [comment]
Hi Team,

The problem I am running into with the `plotProfile` function from `oce` being used for our `plot(argos, which='profile', parameter='x')` is that  because I first need to turn the object into a `ctd` object, the profile MUST have temperature.

I see this being problematic if, for example, a user wants to see the oxygen profile plot for a cycle that does not have temperature.

An example of this is shown below:

```
library(argoFloats)
library(oce)
data("indexBgc")
subset <- subset(indexBgc, 1:10)
argos <- readProfiles(getProfiles(subset))
CTD <- as.ctd(argos)
```
I receive the following error:

```
Error in as.ctd(argos) : must give temperature
```

Am I overthinking this?

**Comment by dankelley** [comment]
You're doing the right amount of thinking!

Indeed, a ctd object *must* have temperature, because otherwise it makes no sense.  But you can always code at a lower level, e.g.
```R
library(oce)
a <- new("ctd")
n <- 100 # for fakes
a <- oceSetData(a, "bark", 1:n+rnorm(n))
a <- oceSetData(a, "pressure", 1:n)
plotProfile(a, xtype="bark")
```

In our case, "bark" would be just the thing the user wants to plot, extracted from the individual `oce:argo-class` objects that are contained in `x@data$argos`, and of course pressure is the pressure in those `oce:argo-class` objects.

I think the best is to create a list of such ctd objects, using `lapply`.

**Comment by j-harbin** [comment]
Right now in  branch`issue241`, when I do the following code:

```R
library(argoFloats)
data(indexSynthetic)
subset <- subset(indexSynthetic,1)
argos <- readProfiles(getProfiles(subset))
plot(argos, which='profile', parameter='temperature')
```
it works fine, however, if instead I do:

```R
plot(argos, which='profile', parameter='oxygen')
```
I receive 

```R
 Error in plot.window(...) : need finite 'ylim' values 
```
It seems this only happens when the y value is not "core" ie. salinity or temperature.

Thoughts?

**Comment by dankelley** [comment]
Want to Z on this?

**Comment by j-harbin** [comment]
As discovery in a vftf with @dankelley, this was a bug within oce and has since been fixed. This issue was solved in commit 7b9145aeafe72abe0c8d1dc88a3eb1cf1e6f7c59 of the develop branch.
